src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { NextFunction, Request, Response } from "express";
+
+import { logger } from "../../utils/logger";
+
+import * as config from "../../config";
+import {
+  BeneficialOwnerTypeChoice,
+  BeneficialOwnerTypeKey,
+} from "../../model/beneficial.owner.type.model";
+import { getApplicationData } from "../../utils/application.data";
+import { ApplicationData } from "../../model";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    return res.render(config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE, {
+      backLinkUrl: config.OVERSEAS_ENTITY_REVIEW_URL,
+      templateName: config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE,
+      ...appData
+    });
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const post = (req: Request, res: Response) => {
+  logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+  return res.redirect(getNextPage(req.body[BeneficialOwnerTypeKey]));
+};
+
+export const postSubmit = (req: Request, res: Response) => {
+  logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+  return res.redirect(config.UPDATE_CHECK_YOUR_ANSWERS_URL);
```

Comment:
This will need to change to go to managing officers type page.

Reply:
Fixed